### ISSUE TITLE
addClaim escapes quotes in value when attempting to write a json value as string

### ISSUE THREAD

**Issue by jeffsteinmetz** [auth0/java-jwt#186]
I am trying to create a JWT similar to an Auth0 token that contains app_metadata, using 3.2.0
is .withClaim escaping quotes?

```
  private val appMetadata = Json.obj("some_key" -> "some_value" ).toString
  private val jwt = JWT.create()
    .withIssuer("https://mydomain.auth0.com/")
    .withClaim("app_metadata", appMetadata)
    .sign(Algorithm.HMAC256("someserect"))
```

appMetadata is a string that returns `{"some_key":"some_value"}`
The JWT created escapes the quotes in the JSON
`{\"some_key\":\"some_value\"}`

**Comment by lbalmaceda** [comment]
You said it: appMetadata is a String.
In order to keep the JSON output valid the quotes are escaped. This is the default behavior of the JSON parser (jackson). It would break the syntax otherwise.. 

There's no API to create tokens with Map claims, but could be added in the future. Feel free to contribute or push a PR like this one (abandoned) https://github.com/auth0/java-jwt/pull/164.

**Comment by jeffsteinmetz** [comment]
To confirm, there is no workaround today that would allow creating a JWT that looks like an Auth0 token (with app_metadata & user_metadata)

**Comment by hzalaz** [comment]
@jeffsteinmetz you could convert a map to string, escape it and then add a string claim to the jwt

**Comment by lbalmaceda** [comment]
☝️ that will work. You create a String claim with a JSON value (i.e. serialized by you using jackson) and then when you need to decode it, parse the value back into a map (i.e. deserialize using jackson).

**Comment by jeffsteinmetz** [comment]
Thank you for the pointers.  Seems it wouldn't matter if .addClaims escapes it, or I escape the string that I pass to .withClaim, the encoded JWT will still come back as

```
{
  "iss": "https://mydomain.auth0.com/",
  "app_metadata": "{\"my_key\":\"my_value\"}"
}
```

even using jackson, converting an Object node back to a string, using that string as the value when calling withClaim would result in the same result wouldn't it?

```
  val mapper = new ObjectMapper()
  val rootNode: ObjectNode = mapper.createObjectNode()
  rootNode.put("my_key", "my_value")
  val jsonString = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(rootNode)
  private val jwt = JWT.create()
    .withIssuer("https://mydomain.auth0.com/")
    .withClaim("app_metadata", jsonString)
    .sign(Algorithm.HMAC256("somesecret"))

  println(jwt)
```
If you debug that JWT (via jwt.io), it would be even worse:
```
  "app_metadata": "{\n  \"profile_uuid\" : \"c76789d7-36da-4bc4-bb9e-f9ca38fe3225\"\n}"
}
```

**Comment by hzalaz** [comment]
@jeffsteinmetz Sorry I understood the other way around, that you needed the json like that. If you need a proper serialized object there is no way now (There is the PR #164 mentioned before that was abandoned).

**Comment by ianaz** [comment]
This has apparently been fixed in #239 . Would be nice to see it merged :)
